app/src/main/java/com/github/libretube/api/PlaylistsHelper.kt

```diff
@@ -133,52 +139,69 @@ object PlaylistsHelper {
         }
     }
 
-    suspend fun importPlaylists(playlists: List<ImportPlaylist>) = withContext(Dispatchers.IO) {
-        playlists.map { playlist ->
-            val playlistId = createPlaylist(playlist.name!!)
-            async {
-                playlistId ?: return@async
-                // if logged in, add the playlists by their ID via an api call
-                if (loggedIn) {
-                    addToPlaylist(
-                        playlistId,
-                        *playlist.videos.map {
-                            StreamItem(url = it)
-                        }.toTypedArray(),
-                    )
-                } else {
-                    // if not logged in, all video information needs to become fetched manually
-                    // Only do so with 20 videos at once to prevent performance issues
-                    playlist.videos.mapIndexed { index, id -> id to index }
-                        .groupBy { it.second % 20 }.forEach { (_, videos) ->
-                            videos.map {
-                                async {
-                                    runCatching {
-                                        val stream = RetrofitInstance.api.getStreams(it.first).toStreamItem(
-                                            it.first,
-                                        )
-                                        addToPlaylist(playlistId, stream)
+    suspend fun importPlaylists(playlists: List<PipedImportPlaylist>) =
```

Comment:
Please try to undo these formatting changes since they are not related to your PR really